scripts/segmentation_HALO-20240907a.md

```diff
@@ -0,0 +1,480 @@
+```
+
+```python
+roll = ds["roll"].hvplot()
+roll * tko * ldn * \
+ seg1s * seg1e * seg2s * seg2e * seg3s * seg3e * seg4s * seg4e * seg5s * seg5e * seg6s * seg6e * seg7s * seg7e * seg8s * seg8e * seg9s * seg9e * \
+ seg10s * seg10e * seg11s * seg11e * seg12s * seg12e * seg13s * seg13e * seg14s * seg14e * seg15s * seg15e * seg16s * seg16e * seg17s * seg17e * seg18s * seg18e * seg19s * seg19e
+```
+
+## Segments
+
+defined as a tuple of time slice (`start`, `end`) , segment `kind`, `name`, `remarks`.
+
+* in case of irregularities within a circle, 1 sec before the first and after the last sonde are chosen as start and end times
+* use the list of `remarks` to state any deviations, also with respective times
+
+Alternatively, you can also define the segments as dictionaries which also allows to add further attributes to single segments, e.g. a `radius` to a `circle` segment. At the end of the following code block all segments will be normalized by the `parse_segments` function.
+
+```python
+seg1 = (
+    slice("2024-09-07T12:56:12", "2024-09-07T13:22:03"),
+    ["straight_leg", "ascent"],
+    "eastward ferry leg 1",
+    []
+)
+
+seg2 = (
+    slice("2024-09-07T13:22:03", "2024-09-07T13:53:07"),
+    ["straight_leg"],
+    "eastward ferry leg 2",
+    []
+)
+
+seg3 = (
+    slice("2024-09-07T13:53:44", "2024-09-07T13:55:33"),
+    ["straight_leg"],
+    "eastward ferry leg 3",
+    []
+)
+
+seg4 = (
+    slice("2024-09-07T13:56:21", "2024-09-07T14:27:45"),
+    ["straight_leg"],
+    "approach first circle",
+    []
+)
+
+
+seg5 = (
+    slice("2024-09-07T14:35:21", "2024-09-07T14:38:38"),
+    ["straight_leg"],
+    "leg inside circle",
+    ["searching for developing clouds for specMACS"]
+)
+
+seg6 = (
+    slice("2024-09-07T14:43:35", "2024-09-07T14:51:22"),
+    ["straight_leg"],
+    "leg inside circle",
+    []
+)
+
+seg7 = (
+    slice("2024-09-07T14:54:12", "2024-09-07T15:49:25"),
+    ["circle"],
+    "counterclockwise southern circle",
+    ["14 sondes, 2 last ones with only 23 sec time difference"]
+)
+
+seg8 = (
+    slice("2024-09-07T15:53:28", "2024-09-07T15:55:18"),
+    ["straight_leg", "ec_track"],
+    "northward ec track",
+    []
+)
+
+seg9 = (
+    slice("2024-09-07T15:59:33", "2024-09-07T17:01:03"),
+    ["circle"],
+    "counterclockwise middle circle",
+    ["irregularities: permission denied for some sondes which were then dropped on straight leg through circle instead"]
+)
+
+seg10 = (
+    slice("2024-09-07T17:05:28", "2024-09-07T17:08:09"),
+    ["straight_leg", "ascent"],
+    "leg back to ec track",
+    []
+)
+
+seg11 = (
+    slice("2024-09-07T17:10:25", "2024-09-07T17:36:23"),
+    ["straight_leg", "ec_track"],
+    "leg through circle",
+    ["sondes dropped inside circle"]
+)
+
+seg12 = (
+    slice("2024-09-07T17:51:40", "2024-09-07T18:59:37"),
+    ["circle"],
+    "clockwise northern circle",
+    ["more than full circle"]
+)
```

Comment:
Depending on our final definition for a `circle` we might want to potentially shorten this segment to only cover the necessary 360deg. But I'd leave it for now.